apps/wpcom-block-editor/src/wpcom/features/live-preview/hooks/use-override-save-button.ts

```diff
@@ -0,0 +1,116 @@
+import { useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { useEffect } from 'react';
+import { getUnlock } from '../utils';
+
+const SAVE_HUB_SAVE_BUTTON_SELECTOR = '.edit-site-save-hub__button';
+const HEADER_SAVE_BUTTON_SELECTOR = '.edit-site-save-button__button';
+
+const unlock = getUnlock();
+
+/**
+ * This overrides the `SaveButton` behavior by adding a listener and changing the copy.
+ * Our objective is to open a custom modal ('ThemeUpgradeModal') instead of proceeding with the default behavior.
+ * For more context, see the discussion on adding an official customization method: https://github.com/WordPress/gutenberg/pull/56807.
+ */
+export const useOverrideSaveButton = ( {
+	setIsThemeUpgradeModalOpen,
+}: {
+	setIsThemeUpgradeModalOpen: any;
+} ) => {
+	const canvasMode = useSelect(
+		( select ) =>
+			unlock && select( 'core/edit-site' ) && unlock( select( 'core/edit-site' ) ).getCanvasMode(),
+		[]
+	);
+
+	useEffect( () => {
+		const saveButtonClickHandler: EventListener = ( e ) => {
+			e.stopPropagation();
+			setIsThemeUpgradeModalOpen( true );
+		};
+		const overrideSaveButtonClick = ( selector: string ) => {
+			const button = document.querySelector( selector );
+			if ( button ) {
+				button.textContent = __( 'Upgrade now', 'wpcom-live-preview' );
+				button.addEventListener( 'click', saveButtonClickHandler );
+			}
+		};
+
+		/**
+		 * This overrides the tooltip text for the save button.
+		 *
+		 * The tooltip is shown after a delay.
+		 * So we observe the DOM changes to avoid being fragile to the delay.
+		 * The observer is activated only when the user hovers over the button to optimize performance.
+		 */
+		const observer = new MutationObserver( ( mutations ) => {
+			mutations.forEach( ( mutation ) => {
+				mutation.addedNodes.forEach( ( node ) => {
+					if ( node.nodeType === Node.ELEMENT_NODE ) {
+						const tooltip = ( node as Element ).querySelector( '.components-tooltip' );
+						if ( tooltip ) {
+							tooltip.textContent = __( 'Upgrade now ⌘S', 'wpcom-live-preview' );
```

Comment:
Can this be something like

```
tooltip.textContent = __( 'Upgrade now', 'wpcom-live-preview' ) + '⌘S';
```

So that it will just reuse the same translation for the Save button?

Reply:
I just checked, and LTR locales will have the command on the left side:
![Screenshot 2023-12-13 at 4 18 48 PM](https://github.com/Automattic/wp-calypso/assets/797888/c0983c3e-3637-4b67-857a-911ba4c0fa29)

Reply:
(╯°□°)╯︵ ┻━┻

Reply:
@taipeicoder @fushar 
I've noticed the command remains on the right side (despite the weird `margin-right`) for the words that have not yet been translated into RTL languages even if I set my language to "العربية". 
<img width="130" alt="Screenshot 2023-12-14 at 12 25 46" src="https://github.com/Automattic/wp-calypso/assets/5287479/4e832f8c-c14b-468a-8796-4d4112032f91">
<img width="334" alt="Screenshot 2023-12-14 at 12 28 29" src="https://github.com/Automattic/wp-calypso/assets/5287479/63bcc469-4cd0-4d73-a10d-b08f59980947">

Can you help me find out how it's implemented (possibly) on the Gutenberg side? 
We may be able to implement it like `textContent = isRtl ? command + text : text + command` as we override the copy anyway, but I want to understand the current implementation to know if we can just use the current DOM structure without doing `textContent = isRtl ? command + text : text + command`. 

Or would it be okay if we take care of RTL in later iterations? What do you think?

Reply:
I found that the shortcut text and its placement is computed dynamically. See:

https://github.com/WordPress/gutenberg/blob/2fb3c57bcadc675f9c6745a99d957b3924e39b71/packages/edit-site/src/components/save-button/index.js#L96

My suggestion for now is to NOT include / leave out the shortcut from the tooltip, due to the complexity. What do you think?

Reply:
> leave out the shortcut from the tooltip

@fushar In that case, the tooltip displays the same text as the button label, making it redundant imo. 😃 
What do you think about hiding the tooltip entirely, then?

Reply:
I'm in favor of leaving out the shortcut, but keeping the tooltip for accessibility purposes. Even though it might be redundant 🙂

Reply:
I addressed this in https://github.com/Automattic/wp-calypso/pull/85013/commits/3af814b75852537d84da08d3f53853af25b4c759 🙂